charts/kubewarden-controller/templates/opentelemetry-collector.yaml

```diff
@@ -9,41 +9,5 @@ metadata:
   annotations:
     {{- include "kubewarden-controller.annotations" . | nindent 4 }}
 spec:
-  mode: sidecar
-  config: 
-    receivers:
-      otlp:
-        protocols:
-          grpc: {}
-    processors:
-      batch: {}
-    exporters:
-      {{- if and .Values.telemetry.tracing.enabled .Values.telemetry.tracing.jaeger.endpoint }}
-      otlp/jaeger:
-        endpoint: {{ .Values.telemetry.tracing.jaeger.endpoint }}
-        {{- if hasKey .Values.telemetry.tracing.jaeger "tls" }}
-        {{- if .Values.telemetry.tracing.jaeger.tls.insecure }}
-        tls:
-          insecure: {{ .Values.telemetry.tracing.jaeger.tls.insecure }}
-        {{- end }}
-        {{- end }}
-      {{- end }}
-      {{- if and .Values.telemetry.metrics.enabled .Values.telemetry.metrics.port }}
-      prometheus:
-        endpoint: ":{{ .Values.telemetry.metrics.port }}"
-      {{- end }}
-    service:
-      pipelines:
-        {{- if and .Values.telemetry.metrics.enabled .Values.telemetry.metrics.port }}
-        metrics:
-          receivers: [otlp]
-          processors: []
-          exporters: [prometheus]
-        {{- end }}
-        {{- if and .Values.telemetry.tracing.enabled .Values.telemetry.tracing.jaeger.endpoint }}
-        traces:
-          receivers: [otlp]
-          processors: [batch]
-          exporters: [otlp/jaeger]
-        {{- end }}
+  {{- toYaml .Values.telemetry.otelSpec | nindent 2 }}
```

Comment:
@jvanz, please don't kill me... but today, while looking into a failure of the e2e tests against this PR we realized everything is a bit more complicated.

Assume this PR is merged. Inside of the `values.yaml` file we have what used to be the previous configuration of the collector. As a user I can do `.Values.telemetry.enabled = true` (we still have this key) and set `.Values.telemetry.tracing.enabled = false`. In this scenario the tracing configuration is still under `.Values.telemetry.otelSpec` unless the user changes it. As a result, the collector would be configured to enable metrics, but it would also have the tracing configuration provided by the default values (which would constantly complain while attempting to send the traces to a non-existing service).

@kravciak, @fabriziosestito and I discussed what to do. Our idea, which we can discuss further with the other maintainers, is to support two scenarios:
- The user enables otel integration with the sidecar model. We take the old code, add a new boolean flag inside of the `Values.yaml` and add some new `if` statements here.
- The user disables the otel integration done with the sidecar and creates on his own otel Collector configuration (maybe create the actual object on his own, without this helm chart?).

We have to do some brainstorming about that...

Reply:
I understand and agree if the finding. By the way, thanks for that. 

But I'm working on the requested changes and wondering if make sense to merge this now. Considering that we need to change the controller to allow Kubewarden to send data to Otel collectors not running in the side car, does make sense to change the Helm chart now?  Our stack only support Otel sidecar now. If we add a flag to disable the sidecar, the telemetry will not work at all. Therefore, any possible customization in Otel collector will not be useful. I think we can park this change, change the controller to allow other Otel deployment models and than came back here an do all the changes at once. Otherwise, if I've apply the request changes now, we will add a flag which will disable sidecar and telemetry will not work. Which is equivalent to what we can do right now by disabling tracing and metrics in the current values structure. 

@kubewarden/kubewarden-developers what do you think?

Reply:
Trying to understand the whole thing.

Ideally I would like to offer the possibility of, via a values key that acts as an enum:
- otel sidecar (current approach).
- an otel collector configuration managed by the chart, that is compatible with stackstate.
- an otel collector configuration inputed by the user, through the chart.

I agree with working on the controller first to allow to send data to otel collectors not on the sidecar, and park this PR.

Reply:
I think we should park this PR, define the whole strategy and then push everything out at the same time.

@jvanz do you want to write a proposal? Maybe this is worth a really lightweight RFC

Reply:
Sorry, I think we can still work on this PR. I was over thinking the goal here is just to add more flexibility in the Otel collector configuration. Not change how the deployment mode. Therefore, I've update the PR changing it to support the old way of configuration. 

If you think we can merge it, fine. Otherwise, we can park it until we fix the controller.

Reply:
I think we should park it as well, I have doubts about the user specifying the spec of the OpenTelemetryCollector by using values.

I would prefer to have either our sidecar "sane defaults" configuration on or entirely off and let the user deal with creating the OpenTelemetryCollector independently from our helm chart.